File: Services/MedicalReport/src/main/java/org/example/medicalreport/Models/entities/GynecologySubRecord.java
package org.example.medicalreport.Models.entities;

import jakarta.persistence.*;
import lombok.*;
import org.example.medicalreport.Models.enums.HormoneStatus;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class GynecologySubRecord extends Sub{
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id; // Assuming an ID field for the entity
    private String work;
    private String civilState;
    private HormoneStatus hormoneStatus;
    private LocalDate menopause;
    private Boolean dysmenorrhea;
    private Boolean menorrhagia;
    private Boolean metrorrhagia;
    private int periodMax;
    private int periodMin;
    private LocalDate date;
    private LocalDate background;

    @OneToMany(mappedBy = "gynecologySubRecord", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Consultation> consultations = new ArrayList<>();

    @ManyToOne
    private MedicalRecord medicalRecord;


}

File: front/src/app/Models/GynecologySubRecord.ts
import { Consultation } from "./Consultation";
import { HormoneStatus } from "./enums";

export class GynecologySubRecord {
  public id!: number;
  public work!: string;
  public civilState!: string;
  public hormoneStatus!: HormoneStatus;
  public menopause!: Date | null;
  public dysmenorrhea!: boolean;
  public menorrhagia!: boolean;
  public metrorrhagia!: boolean;
  public periodMax!: number;
  public periodMin!: number;
  public date!: Date;
  public background!: Date | null;
  public medicalRecordId!: number;
  public consultations!: Consultation[];
}